BuildPropertiesPlugin/README.md

```diff
@@ -0,0 +1,175 @@
+# Android build properties plugin
+**TL;DR** A Gradle plugin to add easy support of property files to your Android build.<br/>
+<br/>
+
+Sometimes you want to retrieve some information from a file that is not checked in as part of your repo for security reasons (keys, credentials, passwords, etc).
+Other times it's just you already are given a properties file and you need to make sure it ends up in your application `BuildConfig` or in some resource file.<br/>
+This plugin augments the `android` default facilities, aiming to provide a simple way to:
+- define handles to properties file in your build script (à la `signingConfig`)
+- generate string fields in your `BuildConfig` with values from a properties file
+- generate string resources with values from a properties file
+- load signing configurations from properties files avoiding to worry about passwords and keystores references in your build script
+
+## How To Build
+
+To build the latest version of the plugin, run from the root of the repo:
+```
+./gradlew build
+```
+
+The plugin at the moment is not deployed in any public maven repo, therefore your only option is to deploy it to a local one.
+A default setup with an embedded repo (pointing to `.gradle/repo`) is provided, just run the following command from the root
+ of the project:
+```
+./gradlew -Plocal uploadArchives
+```
+
+The sample app is not part of the main project to work around the fact the plugin might be not deployed yet either locally or remotely.
+To give it a go you need to **change your working directory to `sample/`**, and build from there. The same `-Plocal` flag is supported by the sample app to use the embedded local repo to retrieve the plugin:
+```
+./gradlew -Plocal assemble
+```
+
+## How To Use
+
+#### 1.Maven coordinates
+The plugin **is not deployed on any public maven repo at the moment**, therefore you have the only choice of using it in
+the sample app via the embedded local repo. The setup for this scenario is already in place, so you just need to specify
+the `-Plocal` flag at build time as explained above (Fear not, Bintray deploy is coming soon).
+
+#### 2. Apply plugin
+The plugin needs to be applied to your Android module after the Android gradle plugin. In your `build.gradle` you should put
+```
+apply plugin: 'build-properties'
+```
+#### 3. List properties files
+In your android project you can add a `buildProperties` configuration listing all the properties files you intend
+to reference in your `android` configuration, eg:
```

Comment:
It actually states you might reference the properties in `buildProperties` configuration in the android configuration (ie: `buildConfigProperty buildProperties.secrets['foo']`)